content/60_gitops_for_socks_shop/30_declare_sock_shop_resource.md

```diff
@@ -0,0 +1,97 @@
++++
+title = "Declare an ArgoCD Helm Chart Object"
+chapter = false
+weight = 30
++++
+
+You'll now declare a [helm chart](https://kubernetes.io/blog/2016/10/helm-charts-making-it-simple-to-package-and-deploy-apps-on-kubernetes/),
+which deploys all of the components that is required to deploy ArgoCD.
+
+Next, let's register an Argo Application in the Kuberletes  chart:
+
+```typescript
+const sockshop = new k8s.apiextensions.CustomResource(
+    "sock-shop",
+    {
+        apiVersion: "argoproj.io/v1alpha1",
+        kind: "Application",
+        metadata: {
+            namespace: "default", // the ns where argocd is deployed
+            name: "sock-shop", // name of app in ArgoCd
+        },
+        spec: {
+            destination: {
+                namespace: ns.metadata.name,
+                server: "https://kubernetes.default.svc",
+            },
+            project: "default",
+            source: {
+                path: "sock-shop",
+                repoURL: "https://github.com/argoproj/argocd-example-apps",
+                targetRevision: "HEAD",
+            },
+            syncPolicy: {
+                automated: {}
+            }
+        }
+    },
+    {provider: provider, parent: ns}
```

Comment:
I don't normally set the namespace as a parent

Reply:
👍